MetaMask/metamask-extension
# Remove sourceUrl from contentscript's inpageBundle

## **Description**

This change makes our contentScript more consistent with this [advice](https://extensionworkshop.com/documentation/develop/build-a-secure-extension/):

> Don't inject moz-extension paths directly
> When injected links, includes, or images include paths to moz-extension://{hash} a page’s tracking script could use this information to [fingerprint](https://en.wikipedia.org/wiki/Device_fingerprint) the user, as the hash (UUID) is unique to the extension installation and, therefore, the user.

more info [here](https://github.com/MetaMask/pm-security/issues/220)

## **Manual testing steps**

All automated tests should pass

We should verify that there is no unexpected behaviour when loading dapps. Try out the metamask test dapp, uniswap and a couple of others. Have the dev console open. Make sure you can connect as expected. Look for unexpected errors in the dev console.

Also include the test case of having a dapp open, closing the browser and then re-opening the browser so that the previously open dapp is the first thing that loads on opening the browser. Verify the dapp loads and connects as expected without unexpected errors (related to MetaMask) in the dev console

An e2e test has been added which should validate that synchronous injection is working.

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
